Source/WebKit/UIProcess/Downloads/DownloadProxy.cpp

```diff
@@ -164,9 +164,13 @@ void DownloadProxy::decideDestinationWithSuggestedFilename(const WebCore::Resour
         completionHandler(destination, WTFMove(sandboxExtensionHandle), allowOverwrite);
 
 #if HAVE(MODERN_DOWNLOADPROGRESS)
-        bool shouldEnableModernDownloadProgress = CFPreferencesGetAppBooleanValue(CFSTR("EnableModernDownloadProgress"), CFSTR("com.apple.WebKit"), nullptr);
-        if (!destination.isEmpty() && shouldEnableModernDownloadProgress)
-            publishProgress(URL::fileURLWithFileSystemPath(destination));
+        if (!destination.isEmpty()) {
+            m_client->decidePlaceholderPolicy(*this, [this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler), destination = WTFMove(destination)] (WebKit::UseDownloadPlaceholder usePlaceholder) mutable {
+                bool shouldEnableModernDownloadProgress = CFPreferencesGetAppBooleanValue(CFSTR("EnableModernDownloadProgress"), CFSTR("com.apple.WebKit"), nullptr);
+                if (shouldEnableModernDownloadProgress)
+                    publishProgress(URL::fileURLWithFileSystemPath(destination), usePlaceholder == WebKit::UseDownloadPlaceholder::Yes);
```

Comment:
I think publishProgress needs to be part of the completion handler.  We shouldn't send the destination to the network process without the placeholder policy.